crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -0,0 +1,367 @@
+			docs: message.docs().join(" "),
+			default: *message.default(),
+		});
+	}
+	Ok(messages)
+}
+
+/// Extracts the information of a smart contract message parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `message` - The label of the contract message.
+fn get_message<P>(path: P, message: &str) -> Result<ContractFunction, Error>
+where
+	P: AsRef<Path>,
+{
+	get_messages(path.as_ref())?
+		.into_iter()
+		.find(|msg| msg.label == message)
+		.ok_or_else(|| Error::InvalidMessageName(message.to_string()))
+}
+
+/// Extracts a list of smart contract contructors parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+pub fn get_constructors(path: &Path) -> Result<Vec<ContractFunction>, Error> {
+	let cargo_toml_path = match path.ends_with("Cargo.toml") {
+		true => path.to_path_buf(),
+		false => path.join("Cargo.toml"),
+	};
+	let contract_artifacts =
+		ContractArtifacts::from_manifest_or_file(Some(&cargo_toml_path), None)?;
+	let transcoder = contract_artifacts.contract_transcoder()?;
+	let mut constructors: Vec<ContractFunction> = Vec::new();
+	for constructor in transcoder.metadata().spec().constructors() {
+		constructors.push(ContractFunction {
+			label: constructor.label().to_string(),
+			payable: *constructor.payable(),
+			args: process_args(constructor.args()),
+			docs: constructor.docs().join(" "),
+			default: *constructor.default(),
+			mutates: true,
+		});
+	}
+	Ok(constructors)
+}
+
+/// Extracts the information of a smart contract constructor parsing the metadata file.
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `constructor` - The label of the constructor.
+fn get_constructor<P>(path: P, constructor: &str) -> Result<ContractFunction, Error>
+where
+	P: AsRef<Path>,
+{
+	get_constructors(path.as_ref())?
+		.into_iter()
+		.find(|c| c.label == constructor)
+		.ok_or_else(|| Error::InvalidConstructorName(constructor.to_string()))
+}
+
+// Parse the parameters into a vector of argument labels.
+fn process_args(params: &[MessageParamSpec<PortableForm>]) -> Vec<Param> {
+	let mut args: Vec<Param> = Vec::new();
+	for arg in params {
+		args.push(Param {
+			label: arg.label().to_string(),
+			type_name: arg.ty().display_name().to_string(),
+		});
+	}
+	args
+}
+
+/// Processes a list of argument values for a specified contract function,
+/// wrapping each value in `Some(...)` or replacing it with `None` if the argument is optional
+///
+/// # Arguments
+/// * `path` -  Location path of the project.
+/// * `label` - Label of the contract message to retrieve.
+/// * `args` - Argument values provided by the user.
+pub fn process_function_args<P>(
+	path: P,
+	label: &str,
+	args: Vec<String>,
+	function_type: FunctionType,
+) -> Result<Vec<String>, Error>
+where
+	P: AsRef<Path>,
+{
+	let parsed_args = args.into_iter().map(|arg| arg.replace(",", "")).collect::<Vec<String>>();
```

Comment:
```suggestion
	let parsed_args = args.into_iter().map(|arg| arg.replace(",", "")).collect::<Vec<_>>();
```
FYI - can use _ to avoid explicitly redefining type.